### PR TITLE
Added longitudinal fit parameters to 7.xx event end fields

### PR DESCRIPTION
**Pull request by AndreaNegro953**
The 74100 file actually has all 0 in the space I am reading but that happens also for the longitudinal chi2 field that was already present in the origin version of pycorsikaio.

### CODE REVIEW

**Comment by AndreaNegro953** [comment]
The test on the units of the event end seems to be the problem, need to figure out why since I had no errors on my machine.

**Comment by maxnoe** [comment]
@AndreaNegro953 The test should fail the same locally. The issue is that `event_end_fields` now is a dict mapping version numbers to the actual fields, not the fields directly anymore.

You need to adapt the test loop over the versions and check all of them.

**Comment by AndreaNegro953** [comment]
@maxnoe Really strange then, I got only warnings about a deprecated thing of numpy, probably I messed something up when running pytest. Anyway I will fix the problem as soon as I can.

File: corsikaio/subblocks/event_end.py [hunk 1]
```diff
@@ -17,4 +19,44 @@
...
+def warn_dtype():
+    warnings.warn("Version unknown, using default event end definition dtype")
+    return event_end_dtype
+
+def warn_fields():
+    warnings.warn("Version unknown, using default event end fields definition")
+    return event_end_fields
+
+event_end_fields = defaultdict(warn_fields)
+event_end_fields[6.5] = event_end_fields
```
Comment (maxnoe) [inline corsikaio/subblocks/event_end.py:51]: And then you insert it into itself. This won't work.

A solution would be to call the variable above `event_end_fields_6xx`

File: corsikaio/subblocks/event_end.py [hunk 2]
```diff
@@ -17,4 +19,44 @@
...
+    Field(266, "n_muons_written"),
+    Field(267, "n_em_particles_preshower"),
+ ]
+
 event_end_dtype = build_dtype(event_end_fields)
+event_end_dtype_7xx = build_dtype(event_end_fields_7xx)
+
+def warn_dtype():
+    warnings.warn("Version unknown, using default event end definition dtype")
+    return event_end_dtype
```
Comment (maxnoe) [inline corsikaio/subblocks/event_end.py:44]: I'd suggest to return the *newest* version we now as default, not the oldest.

File: corsikaio/subblocks/event_end.py [hunk 3]
```diff
@@ -17,4 +19,44 @@
...
+
+def warn_fields():
+    warnings.warn("Version unknown, using default event end fields definition")
+    return event_end_fields
```
Comment (maxnoe) [inline corsikaio/subblocks/event_end.py:48]: same here

File: corsikaio/subblocks/event_end.py (hunk 1 above)
Comment (AndreaNegro953) [reply corsikaio/subblocks/event_end.py]: I understood what happened with tests, I was not using -e option in pip install, my bad

**Comment by maxnoe** [comment]
Can you produce a small test file that contains values for these fields and check they are read correctly?

**Comment by AndreaNegro953** [comment]
Of course, maybe for that I need a little bit more time.

**Comment by maxnoe** [comment]
@AndreaNegro953 I was able create a suitable test file.

Should I just add it here to your branch?

File: corsikaio/subblocks/__init__.py [hunk 4]
```diff
@@ -38,8 +38,8 @@ def parse_event_header(event_header_bytes):
     return np.frombuffer(event_header_bytes, dtype=event_header_types[version])
 
 
-def parse_event_end(event_end_bytes):
-    return np.frombuffer(event_end_bytes, dtype=event_end_dtype)
+def parse_event_end(event_end_bytes,version):
+    return np.frombuffer(event_end_bytes, dtype=event_end_types[float(str(version)[:3])])
```
Comment (maxnoe) [inline corsikaio/subblocks/__init__.py:42]: `round(version, 2)`

**Comment by AndreaNegro953** [comment]
Sorry if I did not reply earlier
Thank you for providing a test file, are there other things I should add to this pull request?

**Comment by maxnoe** [comment]
@AndreaNegro953 No, it looks good! Thanks a lot!

**Review by maxnoe — APPROVED** [review]

**Comment by AndreaNegro953** [comment]
@maxnoe Me and @HealthyPear need to use the parameters added in this PR, would be possible to do a minor release to make this commit available?

**Comment by maxnoe** [comment]
@AndreaNegro953 Yes, I was planning to make a new release together with #39, should be out today

**Comment by AndreaNegro953** [comment]
@maxnoe Thank you!

**Comment by maxnoe** [comment]
@AndreaNegro953 corsika 0.4 is on PyPI: https://pypi.org/project/corsikaio/, conda-forge package will come later today

**Pull request merged**